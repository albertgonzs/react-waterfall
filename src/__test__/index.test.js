import React from 'react'
import renderer from 'react-test-renderer'

import createStore from '../'

const config = {
  initialState: {
    count: 0,
  },
  actions: {
    increment: ({ count }) => ({ count: count + 1 }),
  },
}

global.console = { error: jest.fn() }

test('store initialization', () => {
  const store = createStore(config)

  expect(store.Provider).toBeDefined()
  expect(store.connect).toBeDefined()
  expect(store.actions).toBeDefined()
  expect(store.subscribe).toBeDefined()
  expect(store.unsubscribe).toBeDefined()
})

test('render provider with its children', () => {
  const { Provider } = createStore(config)

  const App = () => (
    <Provider>
      <h1>Children</h1>
    </Provider>
  )

  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('call console.error if an action is called before Provider initialization', () => {
  const { actions } = createStore(config)
  actions.increment()
  expect(console.error).toBeCalledWith('<Provider /> is not initialized yet')
})
