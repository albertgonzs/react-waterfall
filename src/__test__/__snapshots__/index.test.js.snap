// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render provider with its children 1`] = `
<h1>
  Children
</h1>
`;
